ocaml/xapi/dbsync_slave.ml

```diff
@@ -44,6 +44,7 @@ let get_my_ip_addr ~__context =
 
 let create_localhost ~__context info =
   let ip = get_my_ip_addr ~__context in
+  let ip = if ip = "" then "127.0.0.1" else ip in
```

Comment:
just realised this is not in test code, but in prod code, should probably fix this another way 

(the issue is the unit tests do not set up the host with an ip address or a netmask)

Reply:
```

Reply:
What's the source that tells us if ipv4 or ipv6 should be used? Is assuming  ipv4 always here safe?

Reply:
I was more getting at the fact that I shouldn't be changing this line for the purpose of a test, as what I'm effectively writing is `let ip = if RUNNING_UNIT_TESTS then "127.0.0.1" else ip in...`. 

The issue that `Net.Interface` is not mocked, so we have bits of code like:

```
let get_primary_ip_addr ~__context iface primary_address_type =
  if iface = "" then
    None
  else
    (* actually get ip address, using Net.Interface *)
```
But now that I have a test which needs to use an IP we are stuck

@psafont  Ip4 is the only option available as a user at the moment, but we have IPv6 support in theory, so we probably shouldn't assume v4